pkg/vendir/fetch/image/imgpkg.go

```diff
@@ -154,11 +156,30 @@ func (t *Imgpkg) RegistryOpts() (registry.Opts, error) {
 		return registry.Opts{}, err
 	}
 
+	var activeKeychains []auth.IAASKeychain
```

Comment:
should this logic be included into imgpkg itself (e.g. within registry.Opts constructor for example). it seems that we are duplicating functionality. wdyt?

Reply:
maybe we could do it with all the logic, I will look into it